agent-inject/agent/annotations.go

```diff
@@ -20,18 +20,41 @@ const (
 	// be set to a true or false value, as parseable by strconv.ParseBool
 	AnnotationAgentInject = "vault.hashicorp.com/agent-inject"
 
-	// AnnotationAgentInjectSecret is the key annotation that configures Vault
+	// (Deprecated) AnnotationAgentInjectSecret is the key annotation that configures Vault
 	// Agent to retrieve the secrets from Vault required by the app.  The name
 	// of the secret is any unique string after "vault.hashicorp.com/agent-inject-secret-",
 	// such as "vault.hashicorp.com/agent-inject-secret-foobar".  The value is the
 	// path in Vault where the secret is located.
+	// Deprecated because an annotation is max 63 chars in kubernetes and also does not
+	// support directory separators "/"
+	// Any attempt to create a longer name or "/" with this annotation will result in the
+	// secret not being pulled, effectively skipping this annotation
+	// This annotation will always store the file in /vaults/secrets/
+	// This annotation force lowercase the selected name eg vault.hashicorp.com/agent-inject-secret-FOOBAR
+	// will realize the secret file in /vaults/secrets/foobar
 	AnnotationAgentInjectSecret = "vault.hashicorp.com/agent-inject-secret"
 
+	// AnnotationAgentInjectLocation is the annotation that contains the valid vault path to query the
+	// secret from. Anything after the final "-" is used as a unique id to match annotation rules together
+	// use in combination with AnnotationAgentInjectFilename
+	// eg. vault.hashicorp.com/agent-inject-location-foobar: "vault/secret/place"
+	//     vault.hashicorp.com/agent-inject-filename-foobar: "directory/containing/the_secret_filename.txt"
+	AnnotationAgentInjectLocation = "vault.hashicorp.com/agent-inject-location"
+
+	// AnnotationAgentInjectFilename is the annotation that contains the filename to create on disk
+	// Anything after the final "-" is used as a unique id to match annotation rules together
+	// use in combination with AnnotationAgentInjectLocation
+	// If you specify a filename with an absolute location like "/tmp/oreo/my_secret.pem"
+	// it will get created there, if not it defaults to using a root directory of "/vault/secrets/""
```

Comment:
`/vault/secrets` is a volume mount, if we are gonna do this we need to create additional volume mounts otherwise the secret won't be available in the main container

Reply:
Yes, of course